### PR TITLE
Bicaws 2219 add notes

### PR DESCRIPTION
**Pull request by emadkaramad**
PR to enable users to add note to a case

- Added `Add Note` page and feature
- Added `addNote` service
- Fixed lint errors and warnings
- Updated `getDummyCourtCase` to pass the data source to the function to fix `too many clients` errors in Cypress

### CODE REVIEW

File: test/util/insertCourtCases.ts [hunk 1]
```diff
@@ -1,100 +1,94 @@
...
-import { InsertResult } from "typeorm"
+import { DataSource, InsertResult } from "typeorm"
 import getDataSource from "../../src/services/getDataSource"
 import DummyAho from "./error_list_aho.json"
 import DummyCourtCase from "./DummyCourtCase"
 
-const getDummyCourtCase = async (overrides?: Partial<CourtCase>): Promise<CourtCase> => {
-  const dataSource = await getDataSource()
-  return dataSource.getRepository(CourtCase).create({
+const getDummyCourtCase = (dataSource: DataSource, overrides?: Partial<CourtCase>): CourtCase =>
```
Comment (tomvaughan77) [inline test/util/insertCourtCases.ts:9]: Just wondering why passing in `dataSource` is needed here? :)

File: src/features/AddNoteForm/AddNoteForm.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { ComponentMeta, ComponentStory } from "@storybook/react"
+import { within } from "@storybook/testing-library"
+import AddNoteForm from "./AddNoteForm"
+
+export default {
+  title: "Features/CourtCase/Notes/Add",
+  component: AddNoteForm
+} as ComponentMeta<typeof AddNoteForm>
+
+export const WhenCaseNotLockedByAnotherUser: ComponentStory<typeof AddNoteForm> = () => (
```
Comment (tomvaughan77) [inline src/features/AddNoteForm/AddNoteForm.stories.tsx:11]: Small thing - but you could write some extra tests here checking that you can actually write a note in, hit the `Add` button, and that it then is saved (rendered on screen correctly)?

Comment (tomvaughan77) [reply src/features/AddNoteForm/AddNoteForm.stories.tsx:11]: Just feels a bit odd to me that a note-taking component doesn't have stories around it taking notes 📓

File: src/services/entities/CourtCase.ts [hunk 3]
```diff
@@ -124,4 +124,11 @@ export default class CourtCase extends BaseEntity {
 
   @OneToMany(() => Note, (note) => note.courtCase)
   notes!: Relation<Note>[]
+
+  isLockedByAnotherUser(username: string) {
+    return (
+      (!!this.errorLockedById && this.errorLockedById !== username) ||
```
Comment (jaskaransarkaria) [inline src/services/entities/CourtCase.ts:130]: I think I've come across this before, in the db we actually use the username to lock the record and not the user id right? Is this something we need/ want to carry on doing?

Comment (jaskaransarkaria) [reply src/services/entities/CourtCase.ts:130]: or if not can `errorLockedById` -> `errorLockedByUsername`

**Review by jaskaransarkaria — APPROVED** [review]

**Review by tomvaughan77 — APPROVED** [review]

**Pull request merged**